artemis-journal/src/main/java/org/apache/activemq/artemis/core/io/aio/AIOSequentialFileFactory.java

```diff
@@ -304,7 +304,7 @@ public void run() {
          try {
             libaioFile.write(position, bytes, buffer, this);
          } catch (IOException e) {
-            callback.onError(-1, e.getMessage());
```

Comment:
NIO should be fixed separately,  I don't want to include change here, since the JIRA is specifically about the bug.  NIO change is an enhancement.